modules/images/regenerate-existing-images/background-process/class-perflab-background-job.php

```diff
@@ -0,0 +1,380 @@
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	const META_KEY_JOB_ERRORS = 'perflab_job_errors';
+
+	/**
+	 * Job status meta key.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	const META_KEY_JOB_STATUS = 'perflab_job_status';
+
+	/**
+	 * Job status for queued jobs.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	const JOB_STATUS_QUEUED = 'perflab_job_queued';
+
+	/**
+	 * Job status for running jobs.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	const JOB_STATUS_RUNNING = 'perflab_job_running';
+
+	/**
+	 * Job status for partially executed jobs.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	const JOB_STATUS_PARTIAL = 'perflab_job_partial';
+
+	/**
+	 * Job status for completed jobs.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	const JOB_STATUS_COMPLETE = 'perflab_job_complete';
+
+	/**
+	 * Job status for failed jobs.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	const JOB_STATUS_FAILED = 'perflab_job_failed';
+
+	/**
+	 * Job ID.
+	 *
+	 * @since n.e.x.t
+	 * @var int
+	 */
+	private $id;
+
+	/**
+	 * Job name.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	private $name;
+
+	/**
+	 * Job data.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $data;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param int $job_id Job ID. Technically this is the ID of a term in the 'background_job' taxonomy.
+	 */
+	public function __construct( $job_id ) {
+		$this->id = absint( $job_id );
+	}
+
+	/**
+	 * Returns the id for the job.
+	 *
+	 * @return int Job ID. Technically this is a term id for `background_job` taxonomy.
+	 */
+	public function get_id() {
+		return $this->id;
+	}
+
+	/**
+	 * Retrieves the job data.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array|null Job data.
+	 */
+	public function get_data() {
+		$this->data = get_term_meta( $this->id, self::META_KEY_JOB_DATA, true );
+
+		return $this->data;
+	}
+
+	/**
+	 * Retrieves the job name.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string Job name.
+	 */
+	public function get_name() {
+		$this->name = get_term_meta( $this->id, self::META_KEY_JOB_NAME, true );
+
+		return $this->name;
+	}
+
+	/**
+	 * Determines whether a job should run for current request or not.
+	 *
+	 * It determines by checking if job is already running or completed.
+	 * It also checks if max number of attempts have been attempted for the job.
+	 * For any such condition, it will return false, else true.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool Flag to tell whether this job should run or not.
+	 */
+	public function should_run() {
+		// If job doesn't exist or completed, return false.
+		if ( ! $this->exists() || $this->is_completed() ) {
+			return false;
+		}
+
+		if ( $this->is_running() ) {
+			return false;
+		}
+
+		/**
+		 * Number of attempts to try to run a job.
+		 * Repeated attempts may be required to run a failed job. Default 3 attempts.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param int $attempts Number of attempts allowed for a job to run.
```

Comment:
As per the `@Param` documentation. We should add `Default XXX` to it.

`@param: Note if the parameter is Optional before the description, and include a period at the end. The description should mention accepted values as well as the default. For example: Optional. This value does something. Accepts ‘post’, ‘term’, or empty. Default empty.`

```suggestion
		 * @param int $attempts Number of attempts allowed for a job to run. Default 3.
```